### ISSUE TITLE
Breaking change in 1.30.0 (Guzzle ClientException not caught anymore)

### ISSUE THREAD

**Issue by guilliamxavier** [microsoftgraph/msgraph-sdk-php#451]
Updating from 1.29.0 to 1.30.0 (automatically by a `composer update`) broke our app. Relevant part:

```php
$graphRequest = $graph->createRequest($method, $endpoint);
try {
    $graphResponse = $graphRequest->execute();
} catch (\GuzzleHttp\Exception\ClientException $clientException) {
    $response = $clientException->getResponse();
    if ($response && $response->getStatusCode() === 400) {
        /* ... special handling of $response->getBody() ... */
    } else {
        throw $clientException; // rethrow
    }
}
```

(with Guzzle 6.5.5). Now, instead of our special handling, we get a PHP Fatal error:
> Uncaught Microsoft\\Graph\\Exception\\GraphException: [0]: Received 400 for call to https://*(redacted)*
> API response: *(redacted)*

Tracked down to #437, which converts *(not really "wraps", despite the naming)* `\GuzzleHttp\Exception\BadResponseException` to `\Microsoft\Graph\Exception\GraphException`. By the way, the fact that you had to modify the existing test https://github.com/microsoftgraph/msgraph-sdk-php/pull/437/commits/2105503c88ef63348c8941d158527be6a7735e42#diff-4ca1ed8036362337fd93a40c298121e56bb3eefeace0254f9ca40f152c2a68f1 proves that there's a BC break (given that GraphException does *not* extend BadResponseException, even less its children ClientException / ServerException).

I think there are ways to "fix" truncated error messages while preserving the exception type. I'll try to put up a PR. 
[AB#9148](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/9148)

**Comment by Ndiritu** [comment]
@guilliamxavier thanks for reporting this and apologies for the breaking change.
Looking into this: I like the proposed solution of extending the Guzzle exception types.

**Comment by Ndiritu** [comment]
@guilliamxavier I've created a PR with some changes. Feel free to give your feedback as well. Thanks.

**Comment by guilliamxavier** [comment]
@Ndiritu: Thanks but I said I was working on it 😕

> I think there are ways to "fix" truncated error messages while preserving the exception type. I'll try to put up a PR.

So I've also created a PR... let's compare I guess

**Comment by Ndiritu** [comment]
Really sorry @guilliamxavier. 

I think the "try" caused me to take quick action considering it's a breaking change and it was not clear when you would do it. 

Happy to compare and build on your ideas😁

**Comment by guilliamxavier** [comment]
I understand 🙂

By the way I'm wondering: Is it really OK to undo Guzzle's truncation *by default?* Wouldn't it better be an option that must be enabled explicitly? 🤔

**Comment by Ndiritu** [comment]
Thanks for understanding.

Interesting thought. My thoughts are that it's ok since truncation seems primarily driven by memory concerns while logging exception messages and in this case I guess we are somewhat confident about the size of Graph APIs response payloads?

Although once we upgrade Guzzle to >=7.2.0 I think we should make the truncation length/limit configurable by  a user.

Any thoughts @SilasKenneth @MIchaelMainer @zengin ?

**Comment by guilliamxavier** [comment]
Thank you for merging the fix, and for releasing [1.30.1](https://github.com/microsoftgraph/msgraph-sdk-php/releases/tag/1.30.1) 🚀
